4.14 GA Cluster Version Operator and Cloud Credential Operator bugs (#66420)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

[id="ocp-4-14-cloud-cred-operator-bug-fixes"]
==== Cloud Credential Operator

* Previously, the Cloud Credential Operator utility (`ccoctl`) used an incorrect Amazon Resource Names (ARN) prefix for AWS GovCloud (US) and AWS China regions. The incorrect ARN prefix caused the `ccoctl aws create-all` command that is used to create AWS resources during installation to fail. This release updates the ARN prefixes to the correct values. (link:https://issues.redhat.com/browse/OCPBUGS-13549[*OCPBUGS-13549*])

> * Previously, security changes to Amazon s3 buckets caused the Cloud Credential Operator utility (`ccoctl`) command that is used to create AWS resources during installation (`ccoctl aws create-all`) to fail. With this release, the `ccoctl` utility is updated to reflect the Amazon s3 security changes. (link:https://issues.redhat.com/browse/OCPBUGS-11671[*OCPBUGS-11671*])

Reviewer: ```suggestion
* Previously, security changes to Amazon S3 buckets caused the Cloud Credential Operator utility (`ccoctl`) command that is used to create AWS resources during installation (`ccoctl aws create-all`) to fail. With this release, the `ccoctl` utility is updated to reflect the Amazon S3 security changes. (link:https://issues.redhat.com/browse/OCPBUGS-11671[*OCPBUGS-11671*])
```

[discrete]
[id="ocp-4-14-cluster-version-operator-bug-fixes"]
==== Cluster Version Operator

> * Previously, the Cluster Version Operator (CVO) did not reconcile SecurityContextConstraints resources as expected. The CVO now properly reconciles SecurityContextConstraints resources towards the state defined in the release image, reverting any unsupported modifications to them.

Reviewer: Looks like the others got marked up but these ones are still backtickless:
```suggestion
* Previously, the Cluster Version Operator (CVO) did not reconcile `SecurityContextConstraints` resources as expected. The CVO now properly reconciles `SecurityContextConstraints` resources towards the state defined in the release image, reverting any unsupported modifications to them.
```

> * Previously, the Cluster Version Operator did not reconcile SecurityContextConstraints resources as expected. Cluster Version Operator now properly reconciles SecurityContextConstraints resources towards the state defined in the release image, reverting any unsupported modifications to them.
> +
> Users who want to upgrade from earlier OCP versions and who operate workloads depending on modified system SecurityContextConstraints resources need to follow the procedure in https://access.redhat.com/solutions/7033949 to make sure their workloads are able to run without modified system SecurityContextConstraints. (link:https://issues.redhat.com/browse/OCPBUGS-19465[*OCPBUGS-19465*])

Reviewer: ```suggestion
* Previously, the Cluster Version Operator (CVO) did not reconcile `SecurityContextConstraints` resources as expected. The CVO now properly reconciles `SecurityContextConstraints` resources towards the state defined in the release image, reverting any unsupported modifications to them.
+
Users who want to upgrade from earlier {product-title} versions and who operate workloads depending on modified system `SecurityContextConstraints` resources must follow the procedure in link:https://access.redhat.com/solutions/7033949[https://access.redhat.com/solutions/7033949] to make sure their workloads are able to run without modified system `SecurityContextConstraint` resources. (link:https://issues.redhat.com/browse/OCPBUGS-19465[*OCPBUGS-19465*])
```

Alternatively, you could call them "`SecurityContextConstraint` resources (SCCs)" for the first instance and then just say "SCCs" for the rest.

> Users who want to upgrade from earlier OCP versions and who operate workloads depending on modified system SecurityContextConstraints resources need to follow the procedure in https://access.redhat.com/solutions/7033949 to make sure their workloads are able to run without modified system SecurityContextConstraints. (link:https://issues.redhat.com/browse/OCPBUGS-19465[*OCPBUGS-19465*])

Reviewer: Probably aware of this, but reminder that https://access.redhat.com/solutions/7033949 is currently in Unpublished state (i.e., not visible to customers) and should be published before 4.14 GA.

* Previously, the Cluster Version Operator did not prioritize likely targets when deciding which conditional update risks to evaluate first.
+

> With this change, it begins doing so, and conditional updates where the risks happen to not apply will become available more quickly after coming to the Operator's attention.(link:https://issues.redhat.com/browse/OCPBUGS-5469[*OCPBUGS-5469*])

Reviewer: ```suggestion
With this change, it begins doing so, and conditional updates where the risks happen to not apply will become available more quickly after coming to the Operator's attention. (link:https://issues.redhat.com/browse/OCPBUGS-5469[*OCPBUGS-5469*])
```

> * Currently, defining a `sysctl` value for a setting with a slash in its name, such as for bond devices, in the `profile` field of a `Tuned` resource might not work. Values with a slash in the `sysctl` option name are not mapped correctly to the `/proc` filesystem. As a workaround, create a `MachineConfig` resource that places a configuration file with the required values in the `/etc/sysctl.d` node directory.(link:https://issues.redhat.com/browse/RHEL-3707[*RHEL-3707*])

Reviewer: ```suggestion
* Currently, defining a `sysctl` value for a setting with a slash in its name, such as for bond devices, in the `profile` field of a `Tuned` resource might not work. Values with a slash in the `sysctl` option name are not mapped correctly to the `/proc` filesystem. As a workaround, create a `MachineConfig` resource that places a configuration file with the required values in the `/etc/sysctl.d` node directory. (link:https://issues.redhat.com/browse/RHEL-3707[*RHEL-3707*])
```

* Currently, due to an issue with Kubernetes, the CPU Manager is unable to return CPU resources from the last pod admitted to a node to the pool of available CPU resources. These resources can be allocated if a subsequent pod is admitted to the node. However, this in turn becomes the last pod, and again, the CPU manager cannot return the resources of this pod to the available pool.
+

> This issue affects the CPU load balancing features because these features depend on the CPU Manager releasing CPUs to the available pool. Consequently, non-guaranteed pods might run with a reduced number of CPUs. As a workaround, schedule a pod with a `best-effort` CPU Manager policy on the affected node. This pod will be the last pod admitted, ensuring that resources are properly released to the available pool.(link:https://issues.redhat.com/browse/OCPBUGS-17792[*OCPBUGS-17792*])

Reviewer: ```suggestion
This issue affects the CPU load balancing features because these features depend on the CPU Manager releasing CPUs to the available pool. Consequently, non-guaranteed pods might run with a reduced number of CPUs. As a workaround, schedule a pod with a `best-effort` CPU Manager policy on the affected node. This pod will be the last pod admitted, ensuring that resources are properly released to the available pool. (link:https://issues.redhat.com/browse/OCPBUGS-17792[*OCPBUGS-17792*])
```
